nixos/modules/services/web-apps/nextcloud.nix

```diff
@@ -374,6 +376,217 @@ in {
         '';
       };
 
+      defaultLocale = mkOption {
+        default = null;
+        type = types.nullOr types.str;
+        example = "de_CH";
+        description = ''
+          Default locale to use when Nextcloud fails to detect the user's
+          locale.
+
+          Further details about this setting and supported locale codes can be found in the
+          <link xlink:href="https://docs.nextcloud.com/server/stable/admin_manual/configuration_server/language_configuration.html">upstream documentation</link>.
+        '';
+      };
+
+      twofactorEnforcement = let
+        twoFactorAuthUpstreamManualUrl = "https://docs.nextcloud.com/server/latest/admin_manual/configuration_user/two_factor-auth.html";
+      in {
+        enable = mkOption {
+          default = null;
+          type = types.nullOr types.bool;
+          example = true;
+          description = ''
+            ${uiClashWarning}
+
+            Whether to enforce two-factor authentication.
+
+            Further details about this setting and supported locale codes can be found in the
+            <link xlink:href="${twoFactorAuthUpstreamManualUrl}">upstream documentation</link>.
+          '';
+        };
+        groups = mkOption {
+          default = null;
+          type = types.nullOr (types.listOf types.str);
+          example = [ "admin" ];
+          description = ''
+            ${uiClashWarning}
+
+            Enforce two-factor authentication for only users in these named
+            groups.  The groups must already exist in the Nextcloud database.
+
+            Further details about this setting and supported locale codes can be found in the
+            <link xlink:href="${twoFactorAuthUpstreamManualUrl}">upstream documentation</link>.
+          '';
+        };
+        excludedGroups = mkOption {
+          default = null;
+          type = types.nullOr (types.listOf types.str);
+          example = [ "guests" ];
+          description = ''
+            ${uiClashWarning}
+
+            Do <emphasis>not</emphasis> enforce two-factor authentication for
+            users in these named groups.  The groups must already exist in the
+            Nextcloud database.
+
+            Further details about this setting and supported locale codes can be found in the
+            <link xlink:href="${twoFactorAuthUpstreamManualUrl}">upstream documentation</link>.
+          '';
+        };
+      };
+
+      mail = let
+        mailUpstreamManualUrl = "https://docs.nextcloud.com/server/stable/admin_manual/configuration_server/email_configuration.html";
+      in {
+        mode = mkOption {
+          default = null;
+          type = types.nullOr (types.enum [ "smtp" "sendmail" "qmail" ]);
```

Comment:
hmm even when specifying `sendmail` here in a Nix expression for nextcloud, the admin UI shows me `smtp`(`config.php` and `override.config.php` are correct though). Any idea what's up here?